src/journal-remote/journal-gatewayd.c

```diff
@@ -191,10 +192,12 @@ static ssize_t request_reader_entries(
                         /* We skipped beyond the end, make sure entries between the cursor and n_skip offset
                          * from it are not returned. */
                         if (r < m->n_skip + 1) {
-                                m->n_skip -= r;
                                 if (m->follow)
-                                        return 0;
-                                return MHD_CONTENT_READER_END_OF_STREAM;
+                                        wait_for_events = true;
+                                else
+                                        return MHD_CONTENT_READER_END_OF_STREAM;
```

Comment:
Not a big matter, but I'd like to invert the condition to reduce indentation:
```c
if (!m->follow)
        return MHD_CONTENT_READER_END_OF_STREAM;
wait_for_events = true;
```
Feel free to ignore. That's not a strong opinion.